projects/aws/bottlerocket-bootstrap/pkg/kubeadm/utils.go

```diff
@@ -104,6 +105,71 @@ func getBootstrapFromJoinConfig(path string) (string, string, error) {
 	return "https://" + serverEndpoint, token, nil
 }
 
+func getLocalApiServerBindPortFromInitConfig(path string) (int, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return 0, errors.Wrap(err, "Error reading kubeadm init config file")
+	}
+	return getLocalApiBindPortFromInitConfigYaml(string(data))
+}
+
+func getLocalApiBindPortFromInitConfigYaml(yamlString string) (int, error) {
+	yamlTrees, err := utils.UnmarshalYamlIntoMaps(yamlString)
+	if err != nil {
+		return 0, err
+	}
+
+	var kubeadmInitData map[string]interface{}
+	for _, yamlTree := range yamlTrees {
+		kind, ok := yamlTree["kind"]
+		if ok && kind == "InitConfiguration" {
+			kubeadmInitData = yamlTree
+		}
+	}
+	if kubeadmInitData == nil {
+		return 0, errors.New("Cannot find InitConfiguration")
+	}
+
+	localAPIEndpoint := kubeadmInitData["localAPIEndpoint"].(map[string]interface{})
+	bindPort := int(localAPIEndpoint["bindPort"].(float64))
```

Comment:
Why assert to `float64` only to convert to `int`?

Reply:
The yaml parser seem to always parse numeric values in the yaml as `float64`. Try asserting it as `int` causes a panic.

Reply:
No worries, thanks for the clarification.